rockcraft/project.py

```diff
@@ -101,6 +101,29 @@ def _validate_build_base(cls, build_base, values):
             build_base = values.get("base")
         return build_base
 
+    @pydantic.validator("parts", pre=True)
+    @classmethod
+    def _add_pebble_part(cls, parts):
```

Comment:
I am conflicted with this, on one side it is nice to see that a user could do something like:

```yaml
parts:
    pebble:
       source-branch: experimental-branch
```

and for the right thing to happen. This is what we do in Snapcraft which adds an external part from something defined elsewhere https://github.com/snapcore/snapcraft/blob/main/snapcraft/parts/lifecycle.py#L105 ... a similar strategy could be used here where if pebble is not defined we add it (it should not fail validation at this point as we control the input).

@cmatsuoka what do you prefer?

Reply:
I thought about this.
The problem here is that we must be very strict about how Pebble gets into the ROCK, since it is our entrypoint. At most, I do agree it would be nice to allow users to pinpoint a certain Pebble version, but I think this requires some additional discussions to consider:
 - what is the "special" part name that we allow users to define, such that we take that into account when injecting our implicit Pebble part?
 - do we actually need to do this via a part? For now, the only thing users should have a saying about, is what version of Pebble to have installed...isn't a "part" an overkill for such a simple setting?

this sounds like something that requires a spec, cause we would be introducing a naming convention for a special part, that would be taken as an input for our implicit Pebble part. I suggest moving this to a different PR, after we've aligned on the right approach

Reply:
Out of curiosity: do we expect users to want to override/configure the ROCK's pebble? Isn't that similar to, I don't know, configuring the base core22 somehow?

Reply:
I think that's the question we need to tackle. My first instinct would be to say that users could, at most, have a say about the Pebble version that is being installed in their ROCK. 

But we must enforce other things like:
 - how and where Pebble is installed
 - the OCI pebble entrypoint
 - the Pebble env variables
(and maybe more)

Reply:
I believe these latter restrictions should come in the form of linters as the system is flexible enough to allow you to

```
override-prime: |
    craftctl prime
    mv|rm <pebble-path> ...
```

Reply:
yep agreed. thus one more reason why such flexibility should probably be addressed separately, in addition to this PR, as an improvement. resolving this thread in favor of https://github.com/canonical/rockcraft/issues/65